---
title: "Comparative Genome-Resolved Analysis of Two Physiologically Distant ANAMMOX Cultures"
author: "Muhammad Ali"
date: "16 nov 2016"
output:
  pdf_document: default
  html_document: default
  word_document: default
---

```{r message=FALSE, warning=FALSE, results='hide', echo=F}
library("mmgenome")
library ("proto")
library ("ggplot2")
load("data.RData")
options(scipen = 8)
```

## Sample overview

### Figure 1: Differential coverage plot of extracted genome bins

Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the four genome bins that were extracted. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, message=FALSE, warning=FALSE, results='hide', echo=F, fig.align='center', fig.width=7, fig.height=7.5}
mmplot(data = d, 
       x = "S1_2", 
       y = "BS_granule", 
       log.x = T, 
       log.y = T, 
       color = "essential", 
       minlength = 5000) +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.01, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom") +
  annotate("text", x = 1000, y = 300, label = "Bin 1\nCa. Brocadia") +
  annotate("text", x = 40, y = 300, label = "Bin 2\nIgnavibacterium") +
  annotate("text", x = 800, y = 0.03, label = "Bin 3\nFimbriimonadales") +
  annotate("text", x = 900, y = 7, label = "Bin 4\nPHOSâˆ’HE36") +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))
```

### Table 1: Basic metagenome stats
Basic stats of the metagenome libraries and assembly. **Number of scaffolds (#)** is the number of scaffolds in the combined metagenome assembly (all four samples). **Mean GC content (%)** is the mean GC content of all scaffolds in the assembly weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (Mbp)** is the total combined length of the metagenome assembly in Mbp. **Length Maximum (bp)** is the length of the largest scaffold in the assembly. **Length mean** is the mean length of scaffolds in the assembly. **Dataset size** is the size of the each individual metagenome library after quality trimming and mapping to the metagenome assembly. 

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3}
stats <- mmstats(d, ncov = 4) 

stats[4,1] <- stats[4,1] / 1000000 
stats[7,1] <- stats[7,1] * stats[4,1] / 1000
stats[8,1] <- stats[8,1] * stats[4,1] / 1000
stats[9,1] <- stats[9,1] * stats[4,1] / 1000
stats[10,1] <- stats[10,1] * stats[4,1] / 1000

stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (Mbp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Dataset size BS granule (Gbp)",
                           "Dataset size 1.2 (Gbp)",
                           "Dataset size 1.4 (Gbp)",
                           "Dataset size 1.6 (Gbp)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 1))



tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2[1:10,], rows= NULL, theme = tt2)
```

## AMX1: Candidatus Brocadia (RBNG01)

### Figure 2: Initial genome extraction (AMX 1)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **AMX1**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, 
       x = "S1_2", 
       y = "BS_granule", 
       log.x = T, 
       log.y = T, 
       color = "essential", 
       minlength = 5000) +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.1, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(677, 1980, 2050, 576),
                  BS_granule  =  c(468, 506, 2330, 2330))
mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel)
```



### Figure 3: Using paired-end network to extract repeats (AMX1)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 4: Final genome cleanup (AMX1)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **Bin 1**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dB, 
       x = "S1_2", 
       y = "BS_granule", 
       log.x = T, 
       log.y = T, 
       color = "essential", 
       minlength = 1000) +
  scale_x_log10(limits = c(10, 15000), breaks = c(10,100,1000, 10000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(10, 15000), breaks = c(10,100,1000, 10000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(355, 1050, 4420, 14000, 5590, 419),
                  BS_granule  =  c(829, 622, 1160, 8420, 10100, 2000))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dC <- mmextract(dB, sel)
```



### Table 2: Genome bin statistics (AMX1)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dC, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dC, assembly = assembly, file = "CP114_Bin1_CaBrocadia.fa")
```



## IGN2: Ignavibacterium (RBNF01)

### Figure 5: Initial genome extraction (IGN2)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **IGN2**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, 
       x = "S1_2", 
       y = "BS_granule", 
       log.x = T, 
       log.y = T, 
       color = "essential", 
       minlength = 5000) +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.1, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  = c(23.1, 30, 50.4, 60, 40.1, 28.2),
                  BS_granule  =  c(60, 100, 110, 71.3, 36.8, 36.2))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel)
```




### Figure 6: Genome binning refinement (IGN2)
Coverage vs. GC plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the second refinement extraction of scaffolds for **Bin 2**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dA, 
       x = "S1_6", 
       y = "gc", 
       log.x = F, 
       log.y = F, 
       color = "essential", 
       minlength = 1000) +
  scale_x_continuous(name = "Coverage 1.6") + 
  scale_y_continuous(name = "GC (%)") +
  scale_size_area(max_size = 20, breaks = c(2000, 10000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_6  =  c(8.47, 9.97, 15.4, 20.8, 18.7, 14, 9.46),
                  gc  =  c(33.3, 30.9, 29.9, 33.6, 37.9, 41.4, 36.9))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 1))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract(dA, sel)
```



### Figure 7: Using paired-end network to extract repeats (IGN2)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dC <- mmextract_network(subset = dB, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dC, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 8: Final genome cleanup (IGN2)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **Bin 2**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dC, 
       x = "S1_2", 
       y = "BS_granule", 
       log.x = T, 
       log.y = T, 
       color = "essential", 
       minlength = 1000) +
  scale_x_log10(limits = c(1, 1000), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(1, 1000), breaks = c(1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(25.3, 25, 199.7, 373, 310, 50.4),
                  BS_granule  =  c(37.9, 74.1, 407, 600, 300, 50.3))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dD <- mmextract(dC, sel)
```



### Table 3: Genome bin statistics (IGN2)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dD, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dD, assembly = assembly, file = "CP114_Bin2_Ignavibacterium.fa")
```



## IGN3: Fimbriimonadales (RBNE01)

### Figure 9: Initial genome extraction (IGN3)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **IGN3**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, 
       x = "S1_2", 
       y = "BS_granule", 
       log.x = T, 
       log.y = T, 
       color = "essential", 
       minlength = 5000) +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.01, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(100.4, 202, 354, 315, 141),
                  BS_granule  =  c(0.0303, 0.0128, 0.017, 0.0567, 0.0842))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel, exclude = "22552")
```



### Figure 10: Using paired-end network to extract repeats (IGN3)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Table 4: Genome bin statistics (IGN3)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dB, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dB, assembly = assembly, file = "CP114_Bin3_Fimbriimonadales.fa")
```



## CHB4: PHOS-HE36 (RBND01)

### Figure 11: Initial genome extraction (CHB4)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **CHB4**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, 
       x = "S1_2", 
       y = "BS_granule", 
       log.x = T, 
       log.y = T, 
       color = "essential", 
       minlength = 5000) +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.1, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(300, 600, 600, 300),
                  BS_granule  =  c(8, 8, 18, 18))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel, exclude = "27398")
```



### Figure 12: Using paired-end network to extract repeats (CHB4)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 13: Final genome cleanup (CHB4)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **CHB4**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dB, 
       x = "S1_2", 
       y = "S1_4", 
       log.x = F, 
       log.y = F, 
       color = "essential", 
       minlength = 1000) +
  scale_x_continuous(limits = c(1, 1000), name = "Coverage 1.2") + 
  scale_y_continuous(limits = c(1, 300), name = "Coverage 1.4") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(300, 271, 800, 800, 500),
                  S1_4  =  c(120, 175, 275, 198, 130))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dC <- mmextract(dB, sel)
```



### Table 5: Genome bin statistics (CHB4)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dC, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dC, assembly = assembly, file = "CP114_Bin4_PHOS-HE36.fa")
```


## BCD5: Bacteroidetes (RBNC01)

### Figure 14: Initial genome extraction (BCD5)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **BCD5**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, x = "S1_2", y = "BS_granule", log.x = T, log.y = T, color = "essential", minlength = 3000)

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(12.7, 14.7, 13.7, 9.5, 6.62, 4.46, 4.24, 6.39, 9.33),
                  BS_granule  =  c(9.88, 7.08, 3.53, 2.64, 2.23, 3.81, 8.71, 11.4, 11.5))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel)
```



### Figure 15: Using paired-end network to extract repeats (BCD5)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
p <- mmplot(data = dA, x = "S1_6", y = "S1_4", log.x = T, log.y = T, color = "essential")
sel <- data.frame(S1_6  =  c(3.89, 3.51, 3.49, 4.33, 6.17, 5.29),
                  S1_4  =  c(3.08, 2.72, 1.98, 1.73, 1.92, 3.13))
dB <- mmextract(dA, sel)
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 16: Final genome cleanup (BCD5)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **BCD5**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dB, 
       x = "S1_2", 
       y = "S1_4", 
       log.x = F, 
       log.y = F, 
       color = "essential", 
       minlength = 1000) +
  scale_x_continuous(limits = c(1, 1000), name = "Coverage 1.2") + 
  scale_y_continuous(limits = c(1, 300), name = "Coverage 1.4") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_6  =  c(11.5, 11.5, 7.83, 3.3, 1.89, 1.5, 2.37, 5.39),
                  S1_4  =  c(10.6, 5.32, 0.999, 0.301, 0.421, 4.14, 11.6, 17.6))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dC <- mmextract(dB, sel)
```



### Table 6: Genome bin statistics (BCD5)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dC, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dC, assembly = assembly, file = "CP114_BCD5.fa")
```


## ATM6: Bacteroidetes (RBNB01)

### Figure 17: Initial genome extraction (ATM6)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **ATM6**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, x = "S1_2", y = "BS_granule", log.x = T, log.y = T, color = "essential")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(192, 157, 139, 140, 197, 268, 298, 270),
                  BS_granule  =  c(0.0708, 0.0452, 0.0246, 0.0183, 0.0147, 0.0229, 0.0419, 0.0638))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel)
```



### Figure 18: Using paired-end network to extract repeats (ATM6)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 19: Final genome cleanup (ATM6)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **ATM6**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dB, x = "S1_2", y = "BS_granule", log.x = T, log.y = T, color = "essential")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(204, 184, 171, 170, 178, 190, 207, 216, 212),
                  BS_granule  =  c(0.0497, 0.0409, 0.0309, 0.0247, 0.0235, 0.0235, 0.025, 0.0344, 0.0484))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dC <- mmextract(dB, sel)
```



### Table 7: Genome bin statistics (ATM6)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dC, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dC, assembly = assembly, file = "CP114_ATM6.fa")
```


## ATM7: Bacteroidetes (RBNB01)

### Figure 20: Initial genome extraction (ATM7)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **ATM7**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, x = "S1_2", y = "BS_granule", log.x = T, log.y = T, color = "essential", minlength = 3000) +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.1, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(58.5, 41.1, 33.2, 32.9, 39.4, 60.7, 78.9, 69),
                  BS_granule  =  c(2.53, 2, 1.31, 0.85, 0.605, 0.629, 1.03, 1.73))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel)
```



### Figure 21: Using paired-end network to extract repeats (ATM7)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 22: Final genome cleanup (ATM7)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **ATM7**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dB, 
       x = "S1_2", 
       y = "S1_4", 
       log.x = F, 
       log.y = F, 
       color = "essential", 
       minlength = 1000) +
  scale_x_continuous(limits = c(1, 1000), name = "Coverage 1.2") + 
  scale_y_continuous(limits = c(1, 300), name = "Coverage 1.4") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_6  =  c(10.5, 13.5, 18.5, 20.4, 20.3, 17.7, 13.5, 10.5, 8.25, 6.63, 7.04, 8.69),
                  S1_2  =  c(56.7, 61.2, 60.4, 54.8, 51.4, 43, 36.7, 35.1, 35.7, 39.8, 47.4, 53.3))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dC <- mmextract(dB, sel)
```



### Table 8: Genome bin statistics (ATM7)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dC, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dC, assembly = assembly, file = "CP114_ATM7.fa")
```


## CFX8: Bacteroidetes (RBMZ01)

### Figure 23: Initial genome extraction (CFX8)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **CFX8**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, x = "S1_2", y = "BS_granule", log.x = T, log.y = T, color = "essential", minlength = 3000) +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.1, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(115, 150, 118, 96.3, 72, 61.6, 75.4),
                  BS_granule  =  c(4.21, 2.42, 1.56, 1.18, 1.13, 2.26, 3.35))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel, exclude = "27398")
```



### Figure 24: Using paired-end network to extract repeats (CFX8)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 25: Final genome cleanup (CFX8)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **CFX8**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dB, 
       x = "S1_2", 
       y = "S1_4", 
       log.x = F, 
       log.y = F, 
       color = "essential", 
       minlength = 1000) +
  scale_x_continuous(limits = c(1, 1000), name = "Coverage 1.2") + 
  scale_y_continuous(limits = c(1, 300), name = "Coverage 1.4") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_6  =  c(23.3, 20.3, 15.6, 13, 11.8, 13.6, 18.2),
                  S1_4  =  c(50.2, 61.6, 53.3, 42.6, 34.2, 30.4, 36.5))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dC <- mmextract(dB, sel)
```



### Table 9: Genome bin statistics (CFX8)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dC, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dC, assembly = assembly, file = "CP114_CFX8.fa")
```



## CFX9: Bacteroidetes (RBMY01)

### Figure 26: Initial genome extraction (CFX9)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **CFX9**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, x = "S1_2", y = "BS_granule", log.x = T, log.y = T, color = "essential", minlength = 3000) +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.1, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(184, 146, 103, 72.7, 50.5, 51.4, 67.6, 100, 147, 191),
                  BS_granule  =  c(18.2, 20.2, 13.1, 8.92, 5.04, 3.67, 3.26, 5.06, 7.25, 12.3))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel, exclude = "27398")
```



### Figure 27: Using paired-end network to extract repeats (CFX9)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 28: Final genome cleanup (CFX9)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **CFX9**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dB, 
       x = "S1_2", 
       y = "S1_4", 
       log.x = F, 
       log.y = F, 
       color = "essential", 
       minlength = 1000) +
  scale_x_continuous(limits = c(1, 1000), name = "Coverage 1.2") + 
  scale_y_continuous(limits = c(1, 300), name = "Coverage 1.4") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_6  =  c(12.7, 9.15, 6.46, 4.57, 3.81, 3.81, 4, 5.55, 6.99, 9.6, 11.8),
                  S1_4  =  c(54, 51.3, 38.4, 31.3, 23.7, 20.7, 17.2, 19.7, 25, 35, 41.3))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dC <- mmextract(dB, sel)
```



### Table 10: Genome bin statistics (CFX9)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dC, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dC, assembly = assembly, file = "CP114_CFX8.fa")
```



## CFX10: Bacteroidetes (RBMX01)

### Figure 29: Initial genome extraction (CFX10)
Differential coverage plot of the assembled metagenomic scaffolds (>5000 bp), highlighting the initial extraction of scaffolds for **CFX10**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = d, x = "S1_2", y = "BS_granule", log.x = T, log.y = T, color = "essential") +
  scale_x_log10(limits = c(1, 2500), breaks = c(1,10,100,1000), name = "Coverage 1.2") + 
  scale_y_log10(limits = c(0.1, 2500), breaks = c(0.01, 0.1, 1,10,100,1000), name = "Coverage BS granule") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_2  =  c(22.6, 18.4, 16.1, 15.2, 15.1, 17.6, 21.3, 22.5),
                  BS_granule  =  c(3.13, 3.33, 2.67, 2.26, 1.58, 1.47, 1.62, 2.56))

mmplot_selection(p, sel) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 5, shape = 19), nrow = 2))  +
  theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")
```

```{r, warning=FALSE, message=FALSE, echo = F}
dA <- mmextract(d, sel)
```



### Figure 30: Using paired-end network to extract repeats (CFX10)
Until now we have just used coverage profiles to extract scaffolds related to our genome of interest. However, some scaffolds might be present in many copies (repeats) and hence have a much higher coverage than the rest of the genome. In addition, some scaffolds will by chance have a slightly different coverage profile than the rest of the genome and thereby also been missed. Here we generate a network plot of scaffolds connected by paired-end reads and use it to extract scaffolds connected to our genome bin of interest.

```{r, warning=FALSE, message=FALSE, echo = F}
dB <- mmextract_network(subset = dA, original = d, network = pe, nconnections = 5, type = "direct")
```

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7}
mmplot_network(data = dB, network = pe, nconnections = 1, color = "essential") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
  theme(legend.position = "bottom")
```



### Figure 31: Final genome cleanup (CFX10)
Differential coverage plot of the assembled metagenomic scaffolds (all scaffolds) , highlighting the final extracted scaffolds for **CFX10**. The size of the circles represent the length of the scaffolds. Colours of the circles represent phylum level taxonomic classification, based on the essential genes identified on the scaffolds, scaffolds with no colour  either contains no essential genes or could not be assigned a phylum level classification. The x and y-axes show the sequencing coverage in the samples (log-scaled).

```{r, warning=FALSE, message=FALSE, echo = F, fig.align='center', fig.width=7, fig.height=7.5}
p <- mmplot(data = dB, 
       x = "S1_2", 
       y = "S1_4", 
       log.x = F, 
       log.y = F, 
       color = "essential", 
       minlength = 1000) +
  scale_x_continuous(limits = c(1, 1000), name = "Coverage 1.2") + 
  scale_y_continuous(limits = c(1, 300), name = "Coverage 1.4") +
  scale_size_area(max_size = 20, breaks = c(10000, 50000, 100000, 250000, 500000), name = "Scaffold Length") +
  scale_color_discrete(name = "Taxonomic Classification") +
theme(axis.line.x = element_line(),
        axis.line.y = element_line(),
        legend.position = "bottom")

#p
#sel <- mmplot_locator(p)

sel <- data.frame(S1_6  =  c(5.22, 4.69, 4.38, 4.64, 5.13, 5.47),
                  S1_4  =  c(5.88, 5.68, 4.84, 4.3, 4.33, 5.05))
mmplot_selection(p, sel)
```

```{r , warning=FALSE, message=FALSE, echo = F}
dC <- mmextract(dB, sel)
```



### Table 11: Genome bin statistics (CFX10)
Basic stats of the extracted genome bin. **Number of scaffolds (#)** is the number of scaffolds in the extracted genome bin. **Mean GC content (%)** is the mean GC content of all scaffolds in the bin weigthed by scaffold length. **N50 (bp)** is a median statistic that indicates that 50% of the entire assembly is contained in scaffolds equal to or larger than this value (bp).**Length Total (bp)** is the total combined length of extracted genome bin. **Length Maximum (bp)** is the length of the largest scaffold in the extracted genome bin. **Length mean** is the mean length of scaffolds in the extracted genome bin. **Coverage (fold)** is the average coverage for all scaffolds in the bin weighted by their length. **Total essential genes (#)** is the total number of identified essential "single copy" genes in the genome bin, **Unique essential genes (#)** is the number of unique essential "single copy" genes. The list of essential "single copy" genes we look for contain 108 genes in total. If the total number of genes is much higher than the number of unique genes, it is likely that the bin contains scaffolds from more than one organism.

```{r, warning=FALSE, message=FALSE, echo = F, fig.height=3.5}
stats <- mmstats(dC, ncov = 4) 
stats2 <- data.frame(A = c("Number of scaffolds (#)", 
                           "Mean GC content (%)",
                           "N50 (bp)",
                           "Length Total (bp)",
                           "Length Maximum (bp)",
                           "Length Mean (bp)",
                           "Coverage BS granule (fold)",
                           "Coverage 1.2 (fold)",
                           "Coverage 1.4 (fold)",
                           "Coverage 1.6 (fold)",
                           "Total essential genes (#)",
                           "Unique essential genes (#)"), 
                     B = stats)
colnames(stats2) <- c("General Stats", "Value")

stats2 <- mutate(stats2, Value = round(Value, 0))

tt2 <- ttheme_default(core=list(fg_params=list(hjust=1, x = 0.95, fontsize = 10)),
                      colhead=list(fg_params=list(fontsize = 12)))

grid.table(stats2, rows= NULL, theme = tt2)
```

```{r, warning=FALSE, message=FALSE, echo = F}
mmexport(data = dC, assembly = assembly, file = "CP114_CFX8.fa")
```


